discovery/db_discovery.go

```diff
@@ -0,0 +1,92 @@
+package discovery
+
+import (
+	"time"
+
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/go-livepeer/core"
+	lpTypes "github.com/livepeer/go-livepeer/eth/types"
+	"github.com/livepeer/go-livepeer/net"
+
+	"github.com/golang/glog"
+)
+
+type DBOrchestratorPoolCache struct {
+	node *core.LivepeerNode
+}
+
+func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
+	if node.Eth == nil {
+		glog.Error("Could not refresh DB list of orchestrators: LivepeerNode nil")
+		return nil
+	}
+
+	ticker := time.NewTicker(1 * time.Hour)
```

Comment:
Maybe make `1 * time.Hour` a constant?

Also, what do you think of making this overridable so we can explicitly emit ticks during testing? See a general example of this pattern: https://github.com/livepeer/go-livepeer/pull/657#discussion_r244917469

Reply:
Definitely on the constant! 

I'm down to try the pattern. Is what you're suggesting adding a context timeout here, like so? `NewDBOrchestratorPoolCache(node *core.LivepeerNode, ctx context.Context)`. In that example, I'm not sure where the `transcodeLoopContext = stubTranscodeLoopContext` gets used. Is `transcodeLoopContext` a global variable that is then overwritten in the tests?

Reply:
We don't necessarily need to pass in a context right now.

Correct that transcodeLoopContext is a global, and the stub is set during the testing function that needs it.

In this case, stubbing the ticker out cleanly seems a bit tricky because because the Ticker is a concrete type rather than an interface. I'm hoping we can explicitly set the tick, eg 

```
// default impl
getTicker = func() {
  return time.Ticker(1 * time.Hour)
}
...

// during testing
st := time.NewTicker(1 * time.Hour)
stubTicker = func() {
  return st
}
getTicker = stubTicker
.... do things that depend on the ticker ...
st.C <- time.Now() // generate a tick
... check things that depend on the ticker ...
```
Let me know if that works.

Reply:
Update - Josh and I are working on testing the ticker! `st.C <- time.Now()` does not work the way we thought it did